File: java/src/main/java/factory/product/SuperTruck.java
package factory.product;

import factory.product.part.Engine;
import factory.product.part.Tire;

public class SuperTruck extends Vehicle {
    public static final String NAME = "Super Truck";

    public SuperTruck() {
        name = NAME;
        engine = Engine.GASOLINE;
        tire = Tire.OFFLOAD;
    }
}

File: src/main/kotlin/factory/product/SuperTruck.kt
package factory.product

import factory.product.part.Engine
import factory.product.part.Tire

class SuperTruck : Vehicle() {
    override var name = NAME
    override var engine: Engine = Engine.GASOLINE
    override var tire: Tire = Tire.OFFLOAD

    companion object {
        const val NAME: String = "Super Truck"
    }
}